### PR TITLE
feat(mobile): web-parity billing dashboard and billing fixes

### PR DESCRIPTION
**Pull request by ZappoMan**
## Summary
- Restructures the mobile dashboard into the same billing demo flow as the web app: annotated sections for metered usage (`UsageStrip`), Feature A gate, collections grid with selection, collection detail (items + summarize + idempotency), and boolean feature tiles, plus env-gated demo plan controls.
- Adds `randomUuid()` for React Native environments where `crypto.randomUUID` is missing so usage RPCs and idempotency keys work on device.
- Improves `@beakerstack/billing` `mapUnknownError` to read PostgREST-style `{ message, code, details }` objects instead of showing `[object Object]`.
- Includes the prior **fix mobile build** work on this branch (Expo/App wiring, mobile billing URLs, `BillingProvider` / native pricing tweaks, and related tests) so the branch merges cleanly with current mobile billing behavior.

## Test plan
- [ ] `cd apps/mobile && npm test` (at least `DashboardScreen` tests)
- [ ] `cd packages/billing && npx vitest run src/errors.test.ts`
- [ ] Smoke: sign in on iOS/Android sim, open Dashboard, simulate AI summarize against local Supabase with migrations applied (`supabase migration up` or `db reset`)
- [ ] Confirm Billing screen still opens plan/checkout URLs as expected after mobile billing URL changes

### CODE REVIEW

File: packages/billing/src/errors.ts [hunk 1]
```diff
@@ -20,6 +20,36 @@ export function billingError(
...
+      if (typeof o['details'] === 'string' && o['details'].trim())
+        parts.push(o['details']);
+      if (typeof o['hint'] === 'string' && o['hint'].trim())
+        parts.push(o['hint']);
+      return parts.join(' · ');
+    }
+  }
+  if (typeof err === 'string') return err;
+  try {
+    return JSON.stringify(err);
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply packages/billing/src/errors.ts]: **Fix:** `messageFromUnknown` now only returns the `JSON.stringify` result when it is actually a `string`, then falls back to `String(err)` (and finally `'Unknown error'` if even that throws). That covers `undefined`, functions, symbols, and stringify edge cases so `mapUnknownError` never calls `toLowerCase` on a non-string.

Pushed in `aadd114`.

File: packages/billing/src/errors.test.ts [hunk 2]
```diff
@@ -40,6 +40,26 @@ describe('mapUnknownError', () => {
...
+  });
+
+  it('falls back to code and details when message is missing', () => {
+    const e = mapUnknownError({
+      code: '42883',
+      details: 'function does not exist',
+    });
+    expect(e.message).toContain('42883');
+    expect(e.message).toContain('function does not exist');
+  });
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply packages/billing/src/errors.test.ts:61]: **Fix:** Added Vitest cases for `mapUnknownError(undefined)` (asserts a string message and `toLowerCase` does not throw) plus `Symbol` and function inputs, alongside the existing PostgREST-object coverage.

Pushed in `aadd114`.

File: apps/mobile/src/screens/DashboardScreen.tsx [hunk 3]
```diff
@@ -445,77 +89,89 @@ function DemoControlsBlock(): ReactElement | null {
...
-      title='Demo controls'
-      demonstrates='billing_demo_simulate_upgrade, billing_demo_reset_usage'
-      description='App-layer only. Not part of @beakerstack/billing. Remove in production.'
-      codeRef='supabase.rpc("billing_demo_simulate_upgrade", …)'
-      demoMode
-    >
-      <Text style={styles.bodyText}>
-        Current plan: {planLoading ? '…' : (plan?.display_name ?? '—')}
+    <View style={[styles.demoCard, styles.cardDemo]}>
+      <View style={styles.badge}>
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply apps/mobile/src/screens/DashboardScreen.tsx:95]: **Fix:** `DemoControlsBlock` now reads `demoMode` first, calls `usePlan` / `useBillingContext` / `useUsage` / local state hooks unconditionally, then returns `null` when `demoMode` is off—so hook order no longer depends on env toggles.

Pushed in `aadd114`.

File: apps/mobile/src/components/dashboard/UsageStrip.tsx [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    return nextFakeAiSummary();
+  }, []);
+
+  const onSimulate = useCallback(async () => {
+    if (exceeded || pending) return;
+    const summaryText = await resolveSummaryText();
+    const key = pendingKey ?? randomUuid();
+    if (!pendingKey) setPendingKey(key);
+    setPending(true);
+    setRecordError(null);
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply apps/mobile/src/components/dashboard/UsageStrip.tsx:78]: **Fix:** `onSimulate` now sets `simulateInFlightRef` + `pending` + idempotency key *before* `await resolveSummaryText()`, and clears the ref in `finally`, so rapid double-taps cannot start parallel summarize flows before React state updates.

Pushed in `aadd114`.

File: apps/mobile/src/components/dashboard/UsageStrip.tsx [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+import {
+  beakerstackBillingConfig,
+  BEAKERSTACK_METER_AI_SUMMARIZE,
+} from '../../billing/beakerstackBillingConfig';
+import { nextFakeAiSummary } from '../../lib/fakeAi';
+import { randomUuid } from '../../lib/randomUuid';
+import type { ActivityEntry } from './types';
+
+const supabaseRpc = supabase as unknown as SupabaseClient;
+
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply apps/mobile/src/components/dashboard/UsageStrip.tsx]: **Fix:** Removed the `SupabaseClient` cast; `UsageStrip` calls the app’s typed `supabase.rpc('billing_record_usage_event', …)` directly so args stay checked against `Database['public']['Functions']`.

Pushed in `aadd114`.

File: apps/mobile/src/components/dashboard/CollectionDetail.tsx [hunk 6]
```diff
@@ -0,0 +1,401 @@
...
+  BEAKERSTACK_METER_AI_SUMMARIZE,
+} from '../../billing/beakerstackBillingConfig';
+import { nextFakeAiSummary } from '../../lib/fakeAi';
+import { randomUuid } from '../../lib/randomUuid';
+import type { DemoCollectionRow } from '../../billing/useDemoCollections';
+import type { ActivityEntry } from './types';
+import { limLabel } from './utils';
+
+const supabaseRpc = supabase as unknown as SupabaseClient;
+
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply apps/mobile/src/components/dashboard/CollectionDetail.tsx]: **Fix:** Same as `UsageStrip`—`CollectionDetail` now uses typed `supabase.rpc(...)` with no broad `SupabaseClient` cast.

Pushed in `aadd114`.
